pkg/kubelet/pod/pod_manager.go

```diff
@@ -323,23 +319,22 @@ func (pm *basicManager) GetUIDTranslations() (podToMirror map[kubetypes.Resolved
 	return podToMirror, mirrorToPod
 }
 
-func (pm *basicManager) getOrphanedMirrorPodNames() []string {
+func (pm *basicManager) getOrphanedMirrorPodNames() ([]string, []int64) {
 	pm.lock.RLock()
 	defer pm.lock.RUnlock()
 	var podFullNames []string
+	var gracePeriods []int64
 	for podFullName := range pm.mirrorPodByFullName {
 		if _, ok := pm.podByFullName[podFullName]; !ok {
 			podFullNames = append(podFullNames, podFullName)
+			if pm.mirrorPodByFullName[podFullName].Spec.TerminationGracePeriodSeconds != nil {
+				gracePeriods = append(gracePeriods, *pm.mirrorPodByFullName[podFullName].Spec.TerminationGracePeriodSeconds)
+			} else {
+				gracePeriods = append(gracePeriods, 0)
+			}
 		}
 	}
-	return podFullNames
-}
-
-func (pm *basicManager) DeleteOrphanedMirrorPods() {
```

Comment:
its not clear to me why we cannot do the following:

- get orphaned mirror pod names and termination grace periods from api server (if exists)
  - delete mirror pod with specified time.
- else
  - delete static pod

Reply:
Can you elaborate on how the above two actions are coordinated ?
Please note that once the yaml is put back, deleting the mirror pod with specified time has to be retracted.

Suppose call to api server deleting the mirror pod with specified time is successful but the cancellation request to api server fails, we would face inconsistency.